[#42414] Enable feature flags by default in the development and test environment

wielinde opened the pull request:
https://community.openproject.org/work_packages/42414

This PR simply sets the default for the setting `:feature_storages_module_active` to `Rails.env.development?`.

I didn't add any test, as I don't really know how to test that. All tests run in the test environment. Maybe it is not necessary to test it?

wielinde committed:
[#42414] Enable feature flags by default in the development environment

https://community.openproject.org/work_packages/42414

ulferts commented:
@wielinde the storage specs are already run behind that feature flag with each spec enabling it for itself which IMO makes sense since the code by default (production) is to be hidden.

I'd rather keep it like this and only enable storages in dev by default.

wielinde commented:
@ulferts If we disable the feature flag during testing, then we will not know if other specs that don't have the feature enabled fail due to code that was introduced behind a feature flag. I believe that we want to find those spots as soon as possible. Enabling feature flags in testing by default will help here. However, it is true that we then might not find spots where we brake "production" as we already asume that a feature is present.

wielinde committed:
Disable storages feature in test env by default,

so that we can ensure that tests for the
production environment are green. In
production the feature is disabled. We
need to be sure, that production is
working if we want to release.

wielinde commented:
After talking to @ulferts we agreed that we do not enable the features by default in the test environment. Otherwise we would not know how stable the production environment is. // CC @cbliard

cbliard commented:
> After talking to @ulferts we agreed that we do not enable the features by default in the test environment. Otherwise we would not know how stable the production environment is. // CC @cbliard

Makes sense to have the test env as  close to production as possible :+1:
